File: ts/project/.sdk/tm/swift/Sources/ProjectNameSDK/core/Spec.swift
// ProjectName SDK - HTTP request specification.

import Foundation

public final class Spec {
  public var parts: VList? = nil
  public var headers: VMap = VMap()
  public var alias: VMap = VMap()
  public var base: String = ""
  public var prefix: String = ""
  public var suffix: String = ""
  public var params: VMap = VMap()
  public var query: VMap = VMap()
  public var step: String = ""
  public var method: String = "GET"
  public var body: Value = .noval
  public var url: String = ""
  public var path: String = ""

  public init(_ specmap: VMap?) {
    guard let m = specmap else { return }
    if let v = m.entries["parts"], let parts = v.asList { self.parts = parts }
    if let v = m.entries["headers"], let h = v.asMap { self.headers = h }
    if let v = m.entries["alias"], let a = v.asMap { self.alias = a }
    if let v = m.entries["base"], let b = v.asString { self.base = b }
    if let v = m.entries["prefix"], let p = v.asString { self.prefix = p }
    if let v = m.entries["suffix"], let sf = v.asString { self.suffix = sf }
    if let v = m.entries["params"], let pm = v.asMap { self.params = pm }
    if let v = m.entries["query"], let q = v.asMap { self.query = q }
    if let v = m.entries["step"], let st = v.asString { self.step = st }
    if let v = m.entries["method"], let mm = v.asString { self.method = mm }
    if let v = m.entries["body"] { self.body = v }
    if let v = m.entries["url"], let u = v.asString { self.url = u }
    if let v = m.entries["path"], let pa = v.asString { self.path = pa }
  }
}

File: ts/project/.sdk/tm/kotlin/core/Spec.kt
package KOTLINPACKAGE.core

/** The resolved HTTP request specification for one operation. */
@Suppress("UNCHECKED_CAST")
class Spec(specmap: Map<String, Any?>? = null) {

  var parts: MutableList<Any?>? = null
  var headers: MutableMap<String, Any?> = linkedMapOf()
  var alias: MutableMap<String, Any?> = linkedMapOf()
  var base: String = ""
  var prefix: String = ""
  var suffix: String = ""
  var params: MutableMap<String, Any?> = linkedMapOf()
  var query: MutableMap<String, Any?> = linkedMapOf()
  var step: String = ""
  var method: String = "GET"
  var body: Any? = null
  var url: String = ""
  var path: String = ""

  init {
    if (specmap != null) {
      var v = specmap["parts"]
      if (v is MutableList<*>) {
        this.parts = v as MutableList<Any?>
      }
      v = specmap["headers"]
      if (v is MutableMap<*, *>) {
        this.headers = v as MutableMap<String, Any?>
      }
      v = specmap["alias"]
      if (v is MutableMap<*, *>) {
        this.alias = v as MutableMap<String, Any?>
      }
      v = specmap["base"]
      if (v is String) {
        this.base = v
      }
      v = specmap["prefix"]
      if (v is String) {
        this.prefix = v
      }
      v = specmap["suffix"]
      if (v is String) {
        this.suffix = v
      }
      v = specmap["params"]
      if (v is MutableMap<*, *>) {
        this.params = v as MutableMap<String, Any?>
      }
      v = specmap["query"]
      if (v is MutableMap<*, *>) {
        this.query = v as MutableMap<String, Any?>
      }
      v = specmap["step"]
      if (v is String) {
        this.step = v
      }
      v = specmap["method"]
      if (v is String) {
        this.method = v
      }
      if (specmap.containsKey("body")) {
        this.body = specmap["body"]
      }
      v = specmap["url"]
      if (v is String) {
        this.url = v
      }
      v = specmap["path"]
      if (v is String) {
        this.path = v
      }
    }
  }
}
